mods/default/craftitems.lua

```diff
@@ -100,7 +108,16 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 	if formname ~= "default:book" then return end
 	local player_name = player:get_player_name()
 	local inv = player:get_inventory()
-	local stack = player:get_wielded_item()
+	if not book_writers[player_name] then return end
+	local wield_index = book_writers[player_name].wield_index
+	local wield_list = player:get_wield_list()
+	local stack = inv:get_stack(wield_list, wield_index)
+	local written = stack:get_name() == "default:book_written"
+	if stack:get_name() ~= "default:book" and not written then
+		-- No book in the wield slot, abort & inform the player
+		minetest.chat_send_player(player_name, S"The book you were writing to mysteriously disappeared.")
```

Comment:
```suggestion
		minetest.chat_send_player(player_name, S("The book you were writing to mysteriously disappeared."))
```

Reply:
Can't we make an exception for one-letter funcs like `S` (translations)?